portality/tasks/sitemap.py

```diff
@@ -0,0 +1,160 @@
+            volumes = models.JournalVolumeToC.list_volumes(j.id)
+            for v in volumes:
+                vol_loc = base_url + "toc/" + j.toc_id + "/" + v
+                vurl = etree.SubElement(urlset, NS + "url")
+                vloc = etree.SubElement(vurl, NS + "loc")
+                vloc.text = vol_loc
+                cf = etree.SubElement(vurl, NS + "changefreq")
+                cf.text = toc_changefreq
+                counter += 1
+
+        # log to the screen
+        job.add_audit_message(u"{x} urls written to sitemap".format(x=counter))
+
+        # save it into the cache directory
+        attachment_name = 'doaj_' + datetime.strftime(datetime.now(), '%Y%m%d_%H%M') + '.xml'
+        out = os.path.join(smdir, attachment_name)
+        tree = etree.ElementTree(urlset)
+        with open(out, "wb") as f:
+            tree.write(f, encoding="UTF-8", xml_declaration=True, pretty_print=True)
+
+        # update the ES record to point to the new file
+        models.Cache.cache_sitemap(attachment_name)
+
+        job.add_audit_message("Sitemap written to cache directory with name {x}".format(x=attachment_name))
+
+        # remove all but the two latest sitemaps
+        sms = [(c, os.path.getmtime(os.path.join(smdir, c)) ) for c in os.listdir(smdir) if c.endswith(".xml")]
+        sorted_sms = sorted(sms, key=itemgetter(1), reverse=True)
+
+        if len(sorted_sms) > 2:
+            for c, lm in sorted_sms[2:]:
+                os.remove(os.path.join(smdir, c))
+
+    def cleanup(self):
+        """
+        Cleanup after a successful OR failed run of the task
+        :return:
+        """
+        pass
+
+    @classmethod
+    def prepare(cls, username, **kwargs):
+        """
+        Take an arbitrary set of keyword arguments and return an instance of a BackgroundJob,
+        or fail with a suitable exception
+
+        :param kwargs: arbitrary keyword arguments pertaining to this task type
+        :return: a BackgroundJob instance representing this task
+        """
+
+        base_url = app.config.get("BASE_URL")
+        if base_url is None:
+            raise BackgroundException("BASE_URL must be set in configuration before we can generate a sitemap")
+
+        cdir = app.config.get("CACHE_DIR")
+        if cdir is None:
+            raise BackgroundException("You must set CACHE_DIR in the config")
+
+        # first prepare a job record
+        job = models.BackgroundJob()
+        job.user = username
+        job.action = cls.__action__
+        return job
+
+    @classmethod
+    def submit(cls, background_job):
+        """
+        Submit the specified BackgroundJob to the background queue
+
+        :param background_job: the BackgroundJob instance
+        :return:
+        """
+        background_job.save()
+        generate_sitemap.schedule(args=(background_job.id,), delay=10)
+
+@main_queue.periodic_task(crontab(month="*", day="1", hour="0", minute="0"))
```

Comment:
I'd say day="*" is needed here.

Separately, if you'd like to keep it doing exactly what it was doing before, you'd have to set hour="8" as well, but you don't have to - I don't have a strong opinion about when to run it during 1 day.